jwst/regtest/test_nirspec_masterbackground.py

```diff
@@ -0,0 +1,122 @@
+        """
+        # Create optional arguments list using rtdata to find artifact paths
+        user_files = [
+            f'{arg}={rtdata.get_data(input_path)}' for arg, input_path in user_files
+        ] if user_files is not None else []
+
+        if options is None:
+            options = []
+
+        # Get the input data
+        if asn:
+            rtdata.get_asn(input_file)
+
+        else:
+            rtdata.get_data(input_file)
+
+        # Set arguments and execute
+        args = ['config/master_background.cfg', rtdata.input, '--save_results=True'] + user_files + options
+        Step.from_cmdline(args)
+
+        return rtdata
+
+    return _set_pipeline_run
+
+
+def test_nirspec_fs_mbkg_user(run_nirspec_mbkg, fitsdiff_default_kwargs):
+    """Run a test for NIRSpec FS data with a user-supplied background file."""
+    rtdata = run_nirspec_mbkg(
+        'nirspec/fs/nrs_sci+bkg_cal.fits',
+        user_files=[('--user_background', 'nirspec/fs/v2_nrs_bkg_user_clean_x1d.fits')]
+    )
+
+    rtdata.output = "nrs_sci+bkg_master_background.fits"
+
+    # Get the truth file
+    rtdata.get_truth(os.path.join("truth/test_nirspec_fs_mbkg_user", "nrs_sci+bkg_master_background.fits"))
+
+    # Compare the results
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+def test_nirspec_ifu_mbkg_user(run_nirspec_mbkg, fitsdiff_default_kwargs):
+    """Test NIRSpec IFU data with a user-supplied background file."""
+    rtdata = run_nirspec_mbkg(
+        'nirspec/ifu/prism_sci_bkg_cal.fits',
+        user_files=[('--user_background', 'nirspec/ifu/prism_bkg_x1d.fits')]
+    )
+
+    rtdata.output = 'prism_sci_bkg_master_background.fits'
+
+    # Get the truth file
+    rtdata.get_truth(os.path.join("truth/test_nirspec_ifu_mbkg_user", 'prism_sci_bkg_master_background.fits'))
+
+    # Compare the results
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+def test_nirspec_mos_mbkg_user(run_nirspec_mbkg, fitsdiff_default_kwargs):
+    """Test NIRSpec MOS data with a user-supplied background file."""
+    rtdata = run_nirspec_mbkg(
+        'nirspec/mos/nrs_mos_sci+bkg_cal.fits',
+        user_files=[('--user_background', 'nirspec/mos/v2_nrs_mos_bkg_x1d.fits')]
+    )
+
+    rtdata.output = "nrs_mos_sci+bkg_master_background.fits"
+
+    # Get the truth file
+    rtdata.get_truth(os.path.join("truth/test_nirspec_mos_mbkg_user", "nrs_mos_sci+bkg_master_background.fits"))
+
+    # Compare the results
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.parametrize(
+    'output_file', ['ifu_prism_source_on_NRS1_master_background.fits', 'ifu_prism_source_off_NRS1_o001_masterbg.fits']
+)
+def test_nirspec_ifu_nod_mbkg(run_nirspec_mbkg, fitsdiff_default_kwargs, output_file):
```

Comment:
Minor nit, but I'm a consistency nut: How 'bout changing the name of the function to "test_nirspec_ifu_mbkg_nod", so that the naming is consistent with the other test functions above.